File: scripts/generate_icons.py
from PIL import Image
import os

root = os.path.join(os.path.dirname(__file__), '..')
public = os.path.join(root, 'public')
img_dir = os.path.join(public, 'image')
src = os.path.join(img_dir, 'logo512.png')
if not os.path.exists(src):
    print('Source image not found:', src)
    raise SystemExit(1)

sizes = [72,96,128,144,152,192,256,384,512]
img = Image.open(src).convert('RGBA')
for s in sizes:
    out = os.path.join(img_dir, f'logo-{s}.png')
    resized = img.resize((s, s), Image.LANCZOS)
    resized.save(out)
    print('Wrote', out)

# create maskable variant for 512 (same image; ideally the artwork should be extended)
maskable_out = os.path.join(img_dir, 'logo-512-maskable.png')
img.resize((512,512), Image.LANCZOS).save(maskable_out)
print('Wrote', maskable_out)

# create favicon.ico with multiple sizes
ico_path = os.path.join(public, 'favicon.ico')
# Pillow will create multiple sizes from the base image
icon_sizes = [(16,16),(32,32),(48,48),(64,64),(128,128),(256,256)]
img.save(ico_path, format='ICO', sizes=icon_sizes)
print('Wrote', ico_path)

File: scripts/generate_icons.js
const sharp = require('sharp');
let pngToIco = require('png-to-ico');
if (typeof pngToIco !== 'function' && pngToIco && typeof pngToIco.default === 'function') pngToIco = pngToIco.default;
const fs = require('fs');
const path = require('path');

const publicDir = path.join(__dirname, '..', 'public');
const imgDir = path.join(publicDir, 'image');
const src = path.join(imgDir, 'logo512.png');

if (!fs.existsSync(src)) {
  console.error('Source image not found:', src);
  process.exit(1);
}

const sizes = [72,96,128,144,152,192,256,384,512];
(async function main(){
  try{
    for (const s of sizes){
      const out = path.join(imgDir, `logo-${s}.png`);
      await sharp(src).resize(s, s).png().toFile(out);
      console.log('Wrote', out);
    }
    const maskableOut = path.join(imgDir, 'logo-512-maskable.png');
    await sharp(src).resize(512,512).png().toFile(maskableOut);
    console.log('Wrote', maskableOut);

    // create favicon.ico from smaller sizes (16..256)
    const icoSizes = [16,32,48,64,128,256];
    // ensure small pngs exist by resizing from src
    for (const s of icoSizes){
      const out = path.join(imgDir, `logo-${s}.png`);
      await sharp(src).resize(s, s).png().toFile(out);
    }
    const pngPaths = icoSizes.map(s => path.join(imgDir, `logo-${s}.png`));
    const icoBuffer = await pngToIco(pngPaths);
    const icoPath = path.join(publicDir, 'favicon.ico');
    fs.writeFileSync(icoPath, icoBuffer);
    console.log('Wrote', icoPath);

  }catch(err){
    console.error(err);
    process.exit(1);
  }
})();
